pypogs/hardware_mounts.py

```diff
@@ -0,0 +1,815 @@
+        except:
+            pass
+        try:
+            self._logger.debug('Destructor finished')
+        except:
+            pass
+
+    @property
+    def debug_folder(self):
+        """pathlib.Path: Get or set the path for debug logging. Will create folder if not
+        existing."""
+        return self._debug_folder
+    @debug_folder.setter
+    def debug_folder(self, path):
+        # Do not do logging in here! This will be called before the logger is set up
+        path = Path(path)
+        if path.is_file():
+            path = path.parent
+        if not path.is_dir():
+            path.mkdir(parents=True)
+        self._debug_folder = path
+
+    @property
+    def state_cache(self):
+        """dict: Get cache with the current state of the Mount. Updates on calls to get_alt_az() and set_rate_alt_az().
+
+        Keys:
+            azi: float, alt: float, azi_rate: float, alt_rate: float
+        """
+        if self.is_init:
+            return self._state_cache
+        else:
+            return None
+
+    @property
+    def name(self):
+        """str: Get or set the name."""
+        return self._name
+    @name.setter
+    def name(self, name):
+        self._logger.debug('Setting name to: '+str(name))
+        self._name = str(name)
+        self._logger.debug('Name set to '+str(self.name))
+
+    @property
+    def model(self):
+        """str: Get or set the device model.
+
+        Supported:
+            - 'celestron' for Celestron NexStar and Orion/SkyWatcher SynScan hand controllers over serial.
+            - 'ascom'     for ASCOM-enabled telescope mounts.
+        - This will determine which hardware interface is used.
+        - Must set before initialising the device and may not be changed for an initialised device.
+        """
+        return self._model
+    @model.setter
+    def model(self, model):
+        self._logger.debug('Setting model to: '+str(model))
+        assert not self.is_init, 'Can not change already intialised device model'
+        model = str(model)
+        assert model.lower() in self._supported_models,\
+                                                'Model type not recognised, allowed: '+str(self._supported_models)
+        self._model = model.lower()
+        self._logger.debug('Model set to '+str(self.model))
+
+    @property
+    def identity(self):
+        """str: Get or set the device and/or input. Model must be defined first.
+
+        - For model *celestron* this can either be a string with the serial port (e.g. 'COM3' on Windows or
+          '/dev/ttyUSB0' on Linux) or an int with the index in the list of available ports to use (e.g. identity=0 i if
+          only one serial device is connected.)
+        - For model *ascom* this can either be left blank to invoke the ASCOM telescope selection menu, or may specify 
+          a specific installed ASCOM driver by name (case sensitive) (e.g. DeviceHub, Celestron, Simulator, SkyWatcher, etc).
```

Comment:
Is there any way of identifying specific hardware resources? Unlikely to have more than one mount connected but still